backend/pkg/filesystem/util.go

```diff
@@ -21,6 +21,13 @@ func isStringInSlice(item string, arr []string) bool {
 	return false
 }
 
+// isHiddenFile returns true if the filename refers to a file that is hidden, i.e.
+// - either the file belongs to a hidden folder (tmp/..hiddenFolder/file)
+// - or the file itself is a hidden file (tmp/data/.hiddenFile)
+func (c *Service) isHiddenFile(filename string) bool {
```

Comment:
Hey @anishrajan25 !
I guess this is not platform-independent.
I'm not a Golang guru, so I had a chat with GPT and after couple sentences I stay on this example:
It uses `filepath.SkipDir` to not walk into hidden folders and recognize the hidden attribute for Windows
```golang
package main

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"
)

func isHidden(path string, info os.FileInfo) bool {
	// Check for UNIX-style hidden files/directories
	if strings.HasPrefix(filepath.Base(path), ".") {
		return true
	}

	// Check for Windows hidden attribute
	if info.Mode()&os.ModeHidden != 0 {
		return true
	}

	return false
}

func main() {
	root := "." // start from the current directory

	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
		if err != nil {
			return err
		}

		// Skip hidden files and directories
		if isHidden(path, info) {
			if info.IsDir() {
				return filepath.SkipDir // skip the entire directory if it's hidden
			}
			return nil // skip the hidden file and move to the next item
		}

		// Your logic to process the file or directory goes here
		fmt.Println(path)

		return nil
	})

	if err != nil {
		fmt.Printf("error walking the path: %v\n", err)
	}
}
```

Reply:
@smatvienko-tb Thanks for pointing this out.
- It definitely makes more sense to skip a directory if it is hidden, rather than exploring each file inside it
- As for the platform-independent part,
    - I check the os package and it does not contain any constant as `ModeHidden`.
    - Hence I explored more and found out about conditional compilation and how to check if a file or directory is hidden on windows OS by using the `syscall` package. [ref](https://freshman.tech/snippets/go/detect-hidden-file/)
    - I will make the changes accordingly to have conditional compilation because the `syscall` package itself uses it and the `FILE_ATTRIBUTE_HIDDEN` constant that is required is only made available if the compilation is done on WINDOWS.

Reply:
I've updated the PR

Reply:
@anishrajan25, good progress!

I guess the new parameter would be nice to see in the config example: https://github.com/redpanda-data/console/blob/master/docs/config/console.yaml 

I recommend making some unit tests to be 100% sure and secure future updates.